tests/dags/providers/provider_api_scripts/test_smithsonian.py

```diff
@@ -48,6 +49,47 @@ def test_get_hash_prefixes_with_len_one():
     assert actual_prefix_list == expect_prefix_list
 
 
+def test_alert_new_unit_codes():
+    unit_code_set = {"a", "b", "c", "d"}
+    sub_prov_dict = {"sub_prov1": {"a", "c"}, "sub_prov2": {"b"}, "sub_prov3": {"e"}}
+
+    assert si.get_new_and_outdated_unit_codes(unit_code_set, sub_prov_dict) == (
+        {"d"},
+        {"e"},
+    )
+
+
+@pytest.mark.parametrize(
+    "new_unit_codes, outdated_unit_codes",
+    [
+        (("d",), ("e",)),
+        (("d",), ()),
+        ((), ("e",)),
+    ],
+)
+def test_validate_unit_codes_from_api_raises_exception(
+    new_unit_codes, outdated_unit_codes
+):
+    with patch.object(
+        si,
+        "get_new_and_outdated_unit_codes",
+        return_value=(set(new_unit_codes), set(outdated_unit_codes)),
```

Comment:
Could `new_unit_codes` and `outdated_unit_codes` be defined as sets in the parametrization?